ts/mains/main_node.ts

```diff
@@ -648,79 +648,37 @@ async function showAbout() {
   aboutWindow?.show();
 }
 
-let debugLogWindow: BrowserWindow | null = null;
-async function showDebugLogWindow() {
-  if (debugLogWindow) {
-    debugLogWindow.show();
-    return;
-  }
-
-  if (!mainWindow) {
-    console.info('debug log needs mainwindow size to open');
-    return;
-  }
-
-  const theme = await getThemeFromMainWindow();
-  const size = mainWindow.getSize();
-  const options = {
-    width: Math.max(size[0] - 100, getDefaultWindowSize().minWidth),
-    height: Math.max(size[1] - 100, getDefaultWindowSize().minHeight),
-    resizable: true,
-    title: locale.messages.debugLog,
-    autoHideMenuBar: true,
-    backgroundColor: classicDark['--background-primary-color'],
-    shadow: true,
-    show: false,
-    modal: true,
-    webPreferences: {
-      nodeIntegration: true,
-      nodeIntegrationInWorker: false,
-      contextIsolation: false,
-      preload: path.join(getAppRootPath(), 'debug_log_preload.js'),
-      nativeWindowOpen: true,
-    },
-    parent: mainWindow,
-  };
-
-  debugLogWindow = new BrowserWindow(options);
-
-  captureClicks(debugLogWindow);
-
-  await debugLogWindow.loadURL(prepareURL([getAppRootPath(), 'debug_log.html'], { theme }));
-
-  debugLogWindow.on('closed', () => {
-    debugLogWindow = null;
-  });
-
-  debugLogWindow.once('ready-to-show', () => {
-    debugLogWindow?.setBackgroundColor(classicDark['--background-primary-color']);
-  });
-
-  // see above: looks like sometimes ready-to-show is not fired by electron
-  debugLogWindow?.show();
-}
-
-async function saveDebugLog(_event: any, logText: any) {
+async function saveDebugLog(_event: any, additionalInfo: string) {
   const options: Electron.SaveDialogOptions = {
     title: 'Save debug log',
-    defaultPath: path.join(app.getPath('desktop'), `session_debug_${Date.now()}.txt`),
+    defaultPath: path.join(
+      app.getPath('desktop'),
+      `session_debug_${new Date().toISOString().replace(/:/g, '_')}.txt`
+    ),
     properties: ['createDirectory'],
   };
 
   try {
     const result = await dialog.showSaveDialog(options);
     const outputPath = result.filePath;
-    console.info(`Trying to save logs to ${outputPath}`);
+    console.info(`[log] Trying to save logs to ${outputPath}`);
     if (result === undefined || outputPath === undefined || outputPath === '') {
       throw Error("User clicked Save button but didn't create a file");
     }
 
-    fs.writeFile(outputPath, logText, err => {
-      if (err) {
-        throw Error(`${err}`);
-      }
-      console.info(`Saved log - ${outputPath}`);
-    });
+    const loggerFilePath = getLoggerFilePath();
+    if (!loggerFilePath) {
+      throw Error('No logger file path');
+    }
+
+    fs.copyFileSync(loggerFilePath, outputPath);
```

Comment:
have you checked how slow that it for a 20MB file by any chance?

Reply:
we might need to use the non sync one

Reply:
Done!